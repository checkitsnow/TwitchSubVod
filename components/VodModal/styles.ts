import styled from 'styled-components';

interface AdsProps {
  isVisible: boolean;
}

export const Container = styled.div`
  margin-top: 2rem;
  position: relative;
  width: 60rem;
  height: 33.75rem;

  .react-player {
    position: absolute;
    top: 0;
    left: 0;

    iframe {
      height: 100vh;
    }
  }

  @media (max-width: 960px) {
    width: 48rem;
    height: 27rem;
  }

  @media (max-width: 768px) {
    width: 30rem;
    height: 16.875rem;
  }

  @media (max-width: 480px) {
    width: 22.5rem;
    height: 12.65625rem;
  }
`;

export const Ads = styled.div<AdsProps>`
  z-index: ${({ isVisible }) => (isVisible ? '10' : '-1')};
  padding: 10% 20%;
  display: flex;
  flex-direction: column;
  position: absolute;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.7);
  overflow: hidden;
  justify-content: center;
  align-items: center;

  a {
    color: #fafafa;
    background: #323b43;
    padding: 3rem 2rem;
    border-radius: 8px;
    transition: background 0.2s ease;

    p {
      margin: 0;
      opacity: 0.4;
      width: 100%;
      text-align: end;
    }

    :hover {
      background: #1b2125;
    }
  }
  @media (max-width: 768px) {
    padding: 0;

    a {
      padding: 2rem 2rem;
      margin-top: 0;
    }
  }

  button {
    top: 5%;
    right: 3%;
    position: absolute;
    padding: 0.1rem 0.3rem;
    border-radius: 5px;
    line-height: 0;
    background: var(--purple);
    color: inherit;
    font-weight: bold;
    display: flex;
    align-items: center;
    transition: background 0.2s ease;

    svg {
      margin-left: 0.5rem;
    }

    :hover {
      background: var(--dark-purple);
    }
  }
`;
