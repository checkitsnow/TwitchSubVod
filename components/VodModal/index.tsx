import React, { useEffect, useMemo, useState } from 'react';
import ReactPlayer from 'react-player';
import { FaWindowClose } from 'react-icons/fa';

import { Container, Ads } from './styles';

const VodModal = ({ videoUrl }: any) => {
  const [showAd, setShowAd] = useState(true);
  useEffect(() => {
    setShowAd(true);
  }, [videoUrl]);

  const renderVodModal = useMemo(() => {
    return (
      <>
        <Ads isVisible={showAd}>
          <a
            href="https://ko-fi.com/pogulive"
            target="_blank"
            rel="noopener noreferrer"
          >
            Hey, these servers aren't free...
            <img
              className="kofi-img"
              width="230px"
              src="https://cdn.ko-fi.com/cdn/kofi5.png?v=2"
              alt="Buy Me a Coffee at ko-fi.com"
            />
            ... if you can, of course :)
          </a>
          <button onClick={() => setShowAd(false)}>
            <FaWindowClose size={32} />
          </button>
        </Ads>

        <ReactPlayer
          url={videoUrl}
          controls
          width="100%"
          height="100%"
          config={{
            file: {
              hlsOptions: {
                xhrSetup: (xhr: any, _url: any) => {
                  xhr.open('GET', _url.replace('unmuted.ts', 'muted.ts'), true);
                },
              },
            },
          }}
        />
      </>
    );
  }, [videoUrl, showAd, setShowAd]);

  return <Container>{renderVodModal}</Container>;
};

export default VodModal;
