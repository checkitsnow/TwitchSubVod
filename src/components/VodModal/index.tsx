import React from 'react';
import ReactPlayer from 'react-player';

import { Container } from './styles';

const VodModal = ({ videoUrl }: any) => {
  return (
    <Container>
      <ReactPlayer url={videoUrl} controls width="100%" height="100%" />
    </Container>
  );
};

export default VodModal;
