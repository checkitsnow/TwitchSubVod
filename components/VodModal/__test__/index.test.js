import React from 'react';
import renderer from 'react-test-renderer';
import VodModal from '..';

it('should render Vod Modal correctly', () => {
  const tree = renderer
    .create(<VodModal videoUrl="https://www.youtube.com/watch?v=d1YBv2mWll0" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
