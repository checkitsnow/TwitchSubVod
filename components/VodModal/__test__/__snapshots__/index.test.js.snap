// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Vod Modal correctly 1`] = `
<div
  className="styles__Container-xygqpp-0 hhwJGI"
>
  <div
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</div>
`;
